File: NotificationIntelligenceEngine/Kotlin/src/commonTest/kotlin/com/nie/TestVectorModels.kt
/**
 * TestVectorModels.kt
 * Notification Intelligence Engine - Kotlin Tests
 *
 * Models for parsing and applying test vectors.
 */

package com.nie

import kotlinx.serialization.Serializable

/**
 * Root structure for the test vectors JSON file.
 */
@Serializable
data class TestVectorFile(
    val description: String,
    val version: String,
    val vectors: List<TestVector>
)

/**
 * A single test vector defining inputs and expected outputs.
 */
@Serializable
data class TestVector(
    val id: String,
    val description: String,
    val timezone: String,
    val referenceTimeISO: String,
    val triggerLeadTimeMinutes: Int,
    val lookaheadDays: Int,
    val events: List<TestEvent>,
    val skipEventIds: List<String>,
    val expected: TestExpected
)

/**
 * An event as defined in a test vector.
 */
@Serializable
data class TestEvent(
    val id: String,
    val startTimeISO: String,
    val endTimeISO: String? = null
)

/**
 * Expected results for a test vector.
 */
@Serializable
data class TestExpected(
    val expectedEventId: String?,
    val expectedDayLabel: String?,
    val expectedTriggerTimeISO: String?,
    val expectedEndTimeISO: String? = null
)

File: NotificationIntelligenceEngine/Swift/Tests/NotificationIntelligenceTests/TestVectorModels.swift
// TestVectorModels.swift
// Notification Intelligence Engine - Swift Tests
//
// Models for parsing and applying test vectors.

import Foundation

/// Root structure for the test vectors JSON file.
struct TestVectorFile: Codable {
    let description: String
    let version: String
    let vectors: [TestVector]

    enum CodingKeys: String, CodingKey {
        case description = "description"
        case version = "version"
        case vectors = "vectors"
    }
}

/// A single test vector defining inputs and expected outputs.
struct TestVector: Codable {
    let id: String
    let description: String
    let timezone: String
    let referenceTimeISO: String
    let triggerLeadTimeMinutes: Int
    let lookaheadDays: Int
    let events: [TestEvent]
    let skipEventIds: [String]
    let expected: TestExpected
}

/// An event as defined in a test vector.
struct TestEvent: Codable {
    let id: String
    let startTimeISO: String
    let endTimeISO: String?
}

/// Expected results for a test vector.
struct TestExpected: Codable {
    let expectedEventId: String?
    let expectedDayLabel: String?
    let expectedTriggerTimeISO: String?
    let expectedEndTimeISO: String?
}
